src/realm.js

```diff
@@ -1534,6 +1534,9 @@ export class Realm {
       if (root instanceof FunctionEnvironmentRecord && func === root.$FunctionObject) {
         return true;
       }
+      if (root === env.environmentRecord) {
```

Comment:
If the loop below would start at `env` and not `env.parent`, then you wouldn't need this special case. Or is it important that this check runs before the next check around created objects that can return `false`? I don't get the full picture of what is being checked here and the significance of ordering.

Reply:
That's a good point, I found we can leverage `env.destroyed` and use that loop and it simplifies the code quite a bit.